packages/manager/modules/billing/src/payment/fidelity/billing-fidelity.controller.js

```diff
@@ -1,142 +1,161 @@
-) => {
-  $scope.fidelityLoading = false;
-  $scope.fidelityAccountLoading = false;
-  $scope.fidelityAccount = null;
-  $scope.currency = null;
-
-  function getItems() {
-    $scope.fidelityLoading = true;
-
-    BillingFidelity.getMovements(
-      BillingdateRangeSelection.dateFrom,
-      BillingdateRangeSelection.dateTo,
-    )
-      .then((movements) => {
-        $scope.tasksId = movements.reverse();
-      })
-      .catch((data) => {
-        $scope.setMessage(
-          $translate.instant('fidelity_get_movements_error'),
-          data.data,
-        );
-        $scope.loaders.tasks = false;
-      })
-      .finally(() => {
-        $scope.fidelityLoading = false;
+export default class {
+  /* @ngInject */
+  constructor(
+    $filter,
+    $scope,
+    $timeout,
+    $translate,
+    BillingFidelity,
+    BillingmessageParser,
+    BillingdateRangeSelection,
+    coreConfig,
+  ) {
+    this.$filter = $filter;
+    this.$scope = $scope;
+    this.$timeout = $timeout;
+    this.$translate = $translate;
+    this.BillingFidelity = BillingFidelity;
+    this.BillingmessageParser = BillingmessageParser;
+    this.BillingdateRangeSelection = BillingdateRangeSelection;
+    this.coreConfig = coreConfig;
+  }
+
+  $onInit() {
+    this.fidelityLoading = false;
+    this.fidelityAccountLoading = false;
+    this.fidelityAccount = null;
+    this.showCustomRangeDate = false;
+    this.currency = null;
+    this.today = moment();
+
+    // Set items count by page
+    this.itemsPerPage = 10;
+    this.tasksId = [];
+    this.tasksDetails = [];
+
+    this.loaders = {
+      tasks: true,
+    };
+
+    this.currency = this.coreConfig.getUser().currency;
+    this.dateFrom = this.BillingdateRangeSelection.dateFrom;
+    this.dateTo = this.BillingdateRangeSelection.dateTo;
+    return this.getFidelityAccount();
+  }
+
+  getMovements({ offset, pageSize }) {
+    return this.BillingFidelity.getMovements(
+      this.BillingdateRangeSelection.dateFrom,
+      this.BillingdateRangeSelection.dateTo,
+    ).then((movements) => ({
+      data: movements
+        .reverse()
+        .slice(offset - 1, offset + pageSize - 1)
+        .map((movementId) => ({
+          movementId,
+        })),
+      meta: {
+        currentOffset: offset,
+        pageCount: Math.ceil(movements.length / pageSize),
+        pageSize,
+        totalCount: movements.length,
+      },
+    }));
+  }
+
+  getMovementDetails({ movementId }) {
+    return this.BillingFidelity.getMovementsDetails(movementId);
+  }
+
+  onBasicDateRangeChange(_dateFrom) {
+    this.showCustomRangeDate = false;
+    this.dateFrom = moment()
+      .subtract(parseInt(_dateFrom, 10), 'month')
+      .startOf('month');
+
+    this.dateTo = moment();
+
+    this.onDateRangeChange(this.dateFrom, this.dateTo);
+  }
+
+  onDateRangeChange(dateFrom, dateTo) {
+    this.fidelityLoading = true;
+    this.BillingdateRangeSelection.dateFrom = dateFrom;
+    this.BillingdateRangeSelection.dateTo = dateTo;
```

Comment:
**suggestion**: Since `BillingdateRangeSelection` is an injectable, could be better to use set function.

Reply:
Thanks for your suggestion.
We will add it via a separate Pull Request (global refactoring).